projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -34,14 +40,33 @@ export default function JetpackPaymentsIntroEdit( { name, clientId, className }
 		}
 	} );
 
-	const renderVariationPicker = () => {
+	const displayVariations = usableVariations.length && registerBlockVariation;
+
+	let instructions = __( "Please select which kind of payment you'd like to add", 'jetpack' );
+	if ( hasPatterns ) {
+		instructions = __( 'Start by choosing one of our suggested layout patterns', 'jetpack' );
+	}
+
+	if ( ! hasInnerBlocks && displayVariations ) {
 		return (
 			<Placeholder
 				icon={ get( blockType, [ 'icon', 'src' ] ) }
 				label={ get( blockType, [ 'title' ] ) }
-				instructions={ __( "Please select which kind of payment you'd like to add.", 'jetpack' ) }
+				instructions={ instructions }
 				className={ className }
 			>
+				{ hasPatterns && (
+					<>
+						<PaymentsIntroPatternPicker
+							onBlockPatternSelect={ blocks => {
+								const clonedBlocks = blocks.map( block => cloneBlock( block ) );
+								replaceBlock( clientId, clonedBlocks );
+							} }
```

Comment:
Maybe we could create a function here instead of using an anonymous function, I think it would improve readability, maybe something like the setVariation function that is defined on line 29:
```
const setPattern  = blocks => {
    const clonedBlocks = blocks.map( block => cloneBlock( block ) );
    replaceBlock( clientId, clonedBlocks );
};
```
or
```
const setPattern  = blocks => replaceBlock( clientId, blocks.map( block => cloneBlock( block ) ) );
```

Reply:
I don't understand how that makes it more readable. It now means I need to move around the file to find the definition of the function. Given the short function body and the small number of properties, I don't think it's a problem.

Reply:
I agree with @dsas, unless we need to reuse the function somewhere else, I find preferable to keep the anonymous function around as long as it's small. We can make it smaller though, but I don't know it that makes it harder to read 🙂 

```suggestion
							onBlockPatternSelect={ blocks => replaceBlock( clientId, blocks.map( cloneBlock ) }
```

Reply:
Fair enough, I think that's a very valid point and I wouldn't be against it (and for larger functions I'd definitely encourage it).

There is another reason though why it's preferable to have anonymous functions like this one: performance. If we move the implementation to a named function within a functional component, it'd be re-computed on every render. However, by keeping the implementation in an anonymous function that's set directly to a prop, the function is computed only when that prop is called/accessed. 

See [https://reactjs.org/docs/handling-events.html](https://reactjs.org/docs/handling-events.html#:~:text=The%20problem%20with%20this%20syntax%20is%20that%20a%20different%20callback%20is%20created%20each%20time%20the%20LoggingButton%20renders.%20In%20most%20cases%2C%20this%20is%20fine.%20However%2C%20if%20this%20callback%20is%20passed%20as%20a%20prop%20to%20lower%20components%2C%20those%20components%20might%20do%20an%20extra%20re%2Drendering.):

> The problem with this syntax is that a different callback is created each time the `LoggingButton` renders. In most cases, this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re-rendering.

Reply:
Thanks for that, I understand exactly what you mean now. I don't think I saw 'setPattern' as that semantic - `setPattern` isn't different enough from selecting a pattern (probably means setPattern could be better named!). We could come up with something better but I don't agree it's quite so necessary for this one liner if you know the context of the block.